core/include/detray/masks/line.hpp

```diff
@@ -161,6 +161,38 @@ class line {
         return {-xy_bound, -xy_bound, -z_bound, xy_bound, xy_bound, z_bound};
     }
 
+    /// @brief Calculates the coordinates of the vertices.
+    ///
+    /// @param bounds the boundary values for this shape.
+    ///
+    /// @returns a container of vertices. If the shape contains
+    /// no vertices an empty container will be returned.
+    template <typename point3_container_t,
+              template <typename, std::size_t> class bounds_t,
+              typename scalar_t, std::size_t kDIM,
+              typename std::enable_if_t<kDIM == e_size, bool> = true>
+    DETRAY_HOST inline point3_container_t local_vertices(
+        const bounds_t<scalar_t, kDIM> &bounds) const {
+        using point3_t = typename point3_container_t::value_type;
+        point3_t v1 = {-bounds[e_cross_section], -bounds[e_cross_section],
```

Comment:
Yeah I think it just needs to be `{0,0,-bounds[e_half_z]}` and `{0,0,bounds[e_half_z]}`

Reply:
I don't know how it is going to be rendered with the line of `{0,0,-bounds[e_half_z]}` and `{0,0,bounds[e_half_z]}`. it could be very ugly. We can think of displaying three points {0,0,-bounds[e_half_z]}, {0,0,0}, and {0,0,bounds[e_half_z]}